app/services/printer-service.js

```diff
@@ -0,0 +1,22 @@
+import Ember from "ember";
+import AjaxPromise from "stock/utils/ajax-promise";
+
+export default Ember.Service.extend({
+  store: Ember.inject.service(),
+  session: Ember.inject.service(),
+
+  allAvailablePrinters() {
+    return this.get("store")
+      .peekAll("printer")
+      .sortBy("id")
+      .map(printer => printer.getProperties("name", "id"));
+  },
+  updateUserDefaultPrinter(printerId) {
+    new AjaxPromise(
```

Comment:
Change it in future PR's. Inherit this file from `APIBase` service to avoid `AjaxPromise` method usage.